### PR TITLE
add pydap-server as multioutput

### PR DESCRIPTION
**Pull request by kmuehlbauer**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@conda-forge-admin, please rerender

**Comment by kmuehlbauer** [comment]
@conda-forge-admin, please rerender

**Comment by kmuehlbauer** [comment]
@conda-forge-admin, please rerender

**Comment by kmuehlbauer** [comment]
@ocefpaf What is the correct procedure for adding the new output? 

It should be announced over at https://github.com/conda-forge/admin-requests. Do we need to do this beforehand, as this now breaks for the validation.

**Comment by ocefpaf** [comment]
I believe this is the first time we are using that admin request. My guess is that you should send the PR there first with a yaml file like https://github.com/conda-forge/admin-requests/blob/main/examples/example-add-feedstock-output.yml.

**Comment by kmuehlbauer** [comment]
@conda-forge-admin, please restart ci

**Comment by kmuehlbauer** [comment]
Thanks @ocefpaf :heart:, I've seen that behaviour before, but forgot a dummy push will work.

**Comment by ocefpaf** [comment]
> Thanks @ocefpaf ❤️, I've seen that behaviour before, but forgot a dummy push will work.

Yeah. That one is quite annoying. Not sure what is going on there.

**Comment by kmuehlbauer** [comment]
Ready for review, I think :smiley:.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,45 +15,68 @@ source:
...
-    - pip
-  commands:
-    - pip check
-    - pydap -h  # [not win]
+  number: 1
 
+outputs:
+  - name: "pydap"
+    build:
+      script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipe/meta.yaml:23]: This is an overkill in most recipes but it is necessary in multiple outputs due to a bug in conda-build that ignore those flags and end up bringing pip packages.

```suggestion
      script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,45 +15,68 @@ source:
...
+        - gunicorn
+        - jinja2
+        - lxml
+        - PasteDeploy
+        - Webob
+    test:
+      imports:
+        - pydap
+      requires:
+        - gunicorn  # [not win]
```
Comment (ocefpaf) [inline recipe/meta.yaml:73]:
```suggestion
        - python ={{ python_min }}
        - gunicorn  # [not win]
```

**Review by ocefpaf — COMMENTED** [review]
This is kind of new but I believe this is how it was defined in the CFEP. See https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -11,45 +15,70 @@ source:
...
+  number: 1
 
+outputs:
+  - name: "pydap"
+    build:
+      script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
+      noarch: generic
+    requirements:
+      host:
+        - {{ min_python }}
```
Comment (kmuehlbauer) [inline recipe/meta.yaml:27]:
```suggestion
        - python >={{ min_python }}
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -11,45 +15,70 @@ source:
...
+        - pip
+      commands:
+        - pip check
+        - pydap -h  # [not win]
+  - name: "pydap-server"
+    build: 
+      noarch: generic
+    requirements:
+      host:
+        - python {{ python_min }}.*
```
Comment (kmuehlbauer) [inline recipe/meta.yaml:56]:
```suggestion
        - python ={{ python_min }}.*
```

**Comment by ocefpaf** [comment]
BTW, that min_python syntax just changed 😬 

See https://github.com/conda-forge/conda-forge.github.io/pull/2360

**Comment by kmuehlbauer** [comment]
@ocefpaf Now I'm a bit confused, the `python_min` in the pinning feedstock is 3.9. But here the minimum value was 3.10. Not sure where to use which value :grimacing:.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,5 +1,9 @@
 {% set name = "pydap" %}
 {% set version = "3.5.1" %}
+{% set min_python = "3.10" %}
```
Comment (ocefpaf) [inline recipe/meta.yaml:3]:
```suggestion
{% set python_min = "3.10" %}
```

**Comment by ocefpaf** [comment]
> @ocefpaf Now I'm a bit confused, the `python_min` in the pinning feedstock is 3.9. But here the minimum value was 3.10. Not sure where to use which value 😬.

It should be 3.10 based on https://github.com/pydap/pydap/blob/3.5.1/pyproject.toml#L32

File: recipe/meta.yaml [hunk 6]
```diff
@@ -11,45 +15,70 @@ source:
...
+    build: 
+      noarch: generic
+    requirements:
+      host:
+        - python ={{ python_min }}.*
+        - setuptools >=64
+        - setuptools-scm >=8
+        - pip
+      run:
+        - {{ min_python }}
```
Comment (ocefpaf) [inline recipe/meta.yaml:61]:
```suggestion
        - python >={{ python_min }}
```

File: recipe/meta.yaml [hunk 7]
```diff
@@ -11,45 +15,70 @@ source:
...
+  number: 1
 
+outputs:
+  - name: "pydap"
+    build:
+      script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation 
+      noarch: generic
+    requirements:
+      host:
+        - python >={{ min_python }}
```
Comment (ocefpaf) [inline recipe/meta.yaml:27]:
```suggestion
        - python {{ python_min }}
```

**Comment by ocefpaf** [comment]
@kmuehlbauer I think I got them all up to code with the latest doc entry. It should pass now.

**Comment by kmuehlbauer** [comment]
Thanks @ocefpaf for your support :+1:, much appreciated.

**Comment by ocefpaf** [comment]
> Thanks @ocefpaf for your support 👍, much appreciated.

Thank you for the PR! This one was in my  "I'll do it tomorrow" list for a few days 😄

**Comment by Mikejmnez** [comment]
Thanks guys!

**Pull request merged**